packages/core/src/services/chains/ProviderProcessor/processStreamText.ts

```diff
@@ -0,0 +1,64 @@
+import { StreamCommonData } from '../../../events/handlers'
+import { AIReturn } from '../../ai'
+import { saveOrPublishProviderLogs } from './saveOrPublishProviderLogs'
+
+async function processResponse({
+  aiResult,
+  commonData,
+}: {
+  commonData: StreamCommonData
+  aiResult: Awaited<AIReturn<'text'>>
+}) {
+  return {
+    documentLogUuid: commonData.documentLogUuid,
+    text: await aiResult.data.text,
+    usage: await aiResult.data.usage,
+    toolCalls: (await aiResult.data.toolCalls).map((t) => ({
+      id: t.toolCallId,
+      name: t.toolName,
+      arguments: t.args,
+    })),
+  }
+}
+
+async function createProviderLogData({
+  response,
+  aiResult,
+  commonData,
+}: {
+  response: Awaited<ReturnType<typeof processResponse>>
+  aiResult: Awaited<AIReturn<'text'>>
+  commonData: StreamCommonData
+}) {
+  return {
+    ...commonData,
+    responseText: await aiResult.data.text,
+    toolCalls: response.toolCalls,
+  }
+}
```

Comment:
What's the difference between `response` and `aiResult`?

Reply:
`response` is used in the streaming send to the UI. And here I reuse the `toolCalls` because is the same. But `response` is derived from `aiResult`

Reply:
Then we can use only `aiResult` and get `response` from there? Otherwise it's redundant really, right?